File: frontend/src/types/mallDashboard.ts
/**
 * 商場管理統計 Dashboard  TypeScript 型別定義
 */

// ── 巡檢 ──────────────────────────────────────────────────────────────────────

export interface FloorInspectionStats {
  floor:           string   // "b1f" | "b2f" | "rf"
  floor_label:     string   // "B1F" | "B2F" | "RF"
  batches:         number
  total_items:     number
  normal_items:    number
  abnormal_items:  number
  pending_items:   number
  unchecked_items: number
  checked_items:   number
  completion_rate: number
  normal_rate:     number
}

export interface InspectionSummary {
  target_date:     string
  total_batches:   number
  total_items:     number
  checked_items:   number
  unchecked_items: number
  abnormal_items:  number
  completion_rate: number
  by_floor:        FloorInspectionStats[]
}

// ── 週期保養 ──────────────────────────────────────────────────────────────────

export interface PMSummary {
  period_month:     string
  total_items:      number
  completed_items:  number
  incomplete_items: number
  overdue_items:    number
  abnormal_items:   number
  completion_rate:  number
}

// ── Dashboard 整體摘要 ─────────────────────────────────────────────────────────

export interface DashboardSummary {
  inspection:   InspectionSummary
  pm:           PMSummary
  generated_at: string
}

// ── 異常 / 待追蹤清單 ─────────────────────────────────────────────────────────

export interface IssueItem {
  id:           string
  issue_date:   string
  issue_type:   'inspection' | 'pm'
  floor:        string
  item_name:    string
  status:       'abnormal' | 'pending' | 'unchecked' | 'overdue'
  status_label: string
  responsible:  string
  note:         string
  batch_id:     string
}

export interface IssueListResponse {
  items: IssueItem[]
  total: number
}

// ── 趨勢資料 ──────────────────────────────────────────────────────────────────

export interface TrendPoint {
  date:            string
  b1f_completion:  number
  b2f_completion:  number
  rf_completion:   number
  b1f_abnormal:    number
  b2f_abnormal:    number
  rf_abnormal:     number
  total_abnormal:  number
  has_data:        boolean
}

export interface DashboardTrend {
  trend: TrendPoint[]
  days:  number
}

// ── 篩選條件 ──────────────────────────────────────────────────────────────────

export interface DashboardFilters {
  startDate: string | null
  endDate:   string | null
  floor:     'all' | 'b1f' | 'b2f' | 'rf'
  issueType: 'all' | 'inspection' | 'pm'
  status:    'all' | 'abnormal' | 'pending' | 'unchecked' | 'overdue'
}

File: backend/app/schemas/mall_dashboard.py
"""
商場管理統計 Dashboard  Pydantic Schemas
"""
from typing import List, Optional
from pydantic import BaseModel


# ── 巡檢 ──────────────────────────────────────────────────────────────────────

class FloorInspectionStats(BaseModel):
    floor:           str    # "b1f" | "b2f" | "rf"
    floor_label:     str    # "B1F" | "B2F" | "RF"
    batches:         int   = 0
    total_items:     int   = 0
    normal_items:    int   = 0
    abnormal_items:  int   = 0
    pending_items:   int   = 0
    unchecked_items: int   = 0
    checked_items:   int   = 0
    completion_rate: float = 0.0
    normal_rate:     float = 0.0


class InspectionSummary(BaseModel):
    target_date:     str
    total_batches:   int   = 0
    total_items:     int   = 0
    checked_items:   int   = 0
    unchecked_items: int   = 0
    abnormal_items:  int   = 0    # abnormal + pending
    completion_rate: float = 0.0
    by_floor:        List[FloorInspectionStats] = []


# ── 週期保養 ──────────────────────────────────────────────────────────────────

class PMSummary(BaseModel):
    period_month:    str
    total_items:     int   = 0
    completed_items: int   = 0
    incomplete_items: int  = 0
    overdue_items:   int   = 0
    abnormal_items:  int   = 0
    completion_rate: float = 0.0


# ── Dashboard 整體摘要 ─────────────────────────────────────────────────────────

class DashboardSummary(BaseModel):
    inspection:   InspectionSummary
    pm:           PMSummary
    generated_at: str


# ── 異常 / 待追蹤清單 ─────────────────────────────────────────────────────────

class IssueItem(BaseModel):
    id:           str
    issue_date:   str
    issue_type:   str   # "inspection" | "pm"
    floor:        str   # "B1F" | "B2F" | "RF" | "商場"
    item_name:    str
    status:       str   # "abnormal" | "pending" | "unchecked" | "overdue"
    status_label: str
    responsible:  str
    note:         str
    batch_id:     str   # 供 deep-link 使用


class IssueListResponse(BaseModel):
    items: List[IssueItem]
    total: int


# ── 趨勢資料 ──────────────────────────────────────────────────────────────────

class TrendPoint(BaseModel):
    date:            str
    b1f_completion:  float = 0.0
    b2f_completion:  float = 0.0
    rf_completion:   float = 0.0
    b1f_abnormal:    int   = 0
    b2f_abnormal:    int   = 0
    rf_abnormal:     int   = 0
    total_abnormal:  int   = 0
    has_data:        bool  = False


class DashboardTrend(BaseModel):
    trend: List[TrendPoint]
    days:  int
